File: Leetcode/two_pointers/27_RemoveElement/Solution.java
class Solution {
    public int removeElement(int[] nums, int val) {
        int firstVal = 0, temp;
        for (int i = 0; i<nums.length; i++){
            if (nums[i]!=val){
                temp = nums[firstVal];
                nums[firstVal] = nums[i];
                nums[i] = temp;
                firstVal++;
            }
        }
        return firstVal;
    }
}

File: Leetcode/two_pointers/27_RemoveElement/Solution.py
class Solution(object):
    def removeElement(self, nums, val):
        """
        :type nums: List[int]
        :type val: int
        :rtype: int
        """
        firstVal = 0
        while firstVal < len(nums) and nums[firstVal]!=val :
            firstVal += 1
        tracker = firstVal+1
        while tracker < len(nums):
            if nums[tracker]!=val:
                nums[firstVal], nums[tracker] = nums[tracker], nums[firstVal]
                firstVal += 1
            tracker += 1
        return firstVal
